### PR TITLE
use `RunContext` more widely

### PR DESCRIPTION
**Pull request by samuelcolvin**
Also add `model` to `RunContext`.

This was part of my new agent handoff work, but I decided to spit it out.

### CODE REVIEW

File: pydantic_ai_slim/pydantic_ai/_result.py [hunk 1]
```diff
@@ -29,25 +29,22 @@ def __post_init__(self):
     async def validate(
         self,
         result: ResultData,
-        deps: AgentDeps,
-        retry: int,
         tool_call: _messages.ToolCallPart | None,
-        messages: list[_messages.ModelMessage],
+        run_ctx: RunContext[AgentDeps],
```
Comment (sydney-runkle) [inline pydantic_ai_slim/pydantic_ai/_result.py]: We call this `ctx` in our examples, should we be consistent and to that here?

Comment (sydney-runkle) [reply pydantic_ai_slim/pydantic_ai/_result.py]: Or maybe we could just fully use `run_context`?

**Review by sydney-runkle — COMMENTED** [review]
Just one quick nit, otherwise lgtm

File: pydantic_ai_slim/pydantic_ai/_result.py (hunk 1 above)
Comment (samuelcolvin) [reply pydantic_ai_slim/pydantic_ai/_result.py]: yes, probably `run_context` internally is good.

**Pull request merged**